test/NinjaTools.FluentMockServer.API.Tests/Downstream/RequestMapperSpec.cs

```diff
@@ -0,0 +1,355 @@
+                .Then(_ => ThenNoErrorIsReturned())
+                .And(_ => ThenHostHeaderIsNotMapped())
+                .BDDfy();
+        }
+        
+        [BddfyTheory]
+        [ClassData(typeof(HttpContentEmailData))]
+        public void MapAsync_Maps_HttpContent(Stream body, HttpContent expected)
+        {
+            this.Given(_ => GivenInputRequestHasValidRequestUri("http://example.org"))
+                .And(_ => GivenTheInputRequestHasMethod(HttpMethod.Post.ToString()))
+                .And(_ => GivenTheInputRequestHasBody(body))
+                .When(_ => WhenMapped())
+                .And(_ => ThenNoErrorIsReturned())
+                .Then(_ => ThenTheMappedContentIs(expected))
+                .BDDfy("mapped HttpContent.");
+        }
+
+
+        private void ThenTheMappedHasRequestUri(Uri requestUri)
+        {
+            _result.Data?.RequestUri.ShouldBe(requestUri);
+        }
+
+
+        private void ThenTheMappedContentIsNull()
+        {
+            _result.Data?.Content.ShouldBeNull();
+        }
+
+
+        private void ThenHostHeaderIsNotMapped()
+        {
+            _result.Data.Headers.Host.ShouldBeNull();
+        }
+
+      
+        private void GivenTheInputRequestHasMethod([NotNull] string method)
+        {
+            _inputRequest.Method = new HttpMethod(method).Method;
+        }
+
+        private void GivenTheInputRequestHasHostWithPort(string host, int? port = null)
+        {
+            _inputRequest.Host = port.HasValue
+                ? new HostString(host, port.Value)
+                : new HostString(host);
+        }
+
+
+        private void GivenTheInputRequestHasNoHeaders()
+        {
+            _inputRequest.Headers.Clear();
+        }
+
+
+        private void GivenTheInputRequestHasPath([CanBeNull] string path)
+        {
+            if (path != null)
+            {
+                _inputRequest.Path = path;
+            }
+        }
+
+        private void GivenTheInputRequestHasQueryString([CanBeNull] string querystring)
+        {
+            if (querystring != null)
+            {
+                _inputRequest.QueryString = new QueryString(querystring);
+            }
+        }
+
+
+        private async Task WhenMapped()
+        {
+            _result = await _requestMapper.MapAsync(_inputRequest);
+
+            if (_result.Data != null && _result.Data is { } data )
+            {
+                _mail = await data.Content.DeserializeAsync<Email>();
+               
+            }
+        }
+
+
+        private void GivenInputRequestHasValidRequestUri([NotNull] string requestUri)
+        {
+            var uri = new Uri(requestUri);
+            _inputRequest.Scheme = uri.Scheme;
+            GivenTheInputRequestHasHostWithPort(uri.Authority, uri.Port);
+        }
+        
+        private void GivenTheInputRequestHasBody([NotNull] Stream stream)
+        {
+            _inputRequest.Body = stream;
+        //    _inputRequest.EnableBuffering();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/app/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4722826)